x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/plugin/SqlPlugin.java

```diff
@@ -158,4 +159,9 @@ protected XPackLicenseState getLicenseState() {
             infoAction
         );
     }
+
+    @Override
+    public List<NamedWriteableRegistry.Entry> getNamedWriteables() {
```

Comment:
What's the advantage of sharing the cursors writables since in most of the code, the StreamInput is wrapped?

Reply:
Not quite sure what you mean by `since in most of the code, the StreamInput is wrapped` but it became necessary because there is no way atm to combine two `NamedWriteableRegistry`. But `SqlStreamInput` needs to be able to decode both instances of `Cursor` and `SearchSourceBuilder` (in `CompositeAggCursor`) of which the later is only available through the global registry.

I think there are not really any major downsides though. Since the `NamedWriteableRegistry` uses fully qualified class names for the writeable "categories", there shouldn't be any visible effect for the other plugins.